### DISCUSSION TITLE
Client Component not working (send help)

### DISCUSSION THREAD

**Discussion by cedricmarcelino** [vercel/next.js#60797] [Help]
### Summary

I am currently learning NextJS and I am having trouble with Client Components using app router. I am trying to use the example provided in the documentation. Here is how I implemented it:

app/layout.tsx
```
import React from 'react'

export default function RootLayout({
    children,
  }: {
    children: React.ReactNode
  }) {
    return (
      <html lang="en">
        <body>{children}</body>
      </html>
    )
  }
```

app/page.tsx
```
import React from 'react'
import Counter from './components/Counter'

export default function Page() {
    return (
        <Counter/>
    )
}
```

app/components/Counter
```
'use client'
 
import { useState } from 'react'
 
export default function Counter() {
  const [count, setCount] = useState(0)
 
  return (
    <div>
      <p>You clicked {count} times</p>
      <button onClick={() => setCount(count + 1)}>Click me</button>
    </div>
  )
}
```

However, I am getting the error "Uncaught TypeError: Cannot read properties of undefined (reading 'call')" in the browser's UI. I've been tinkering with it for a whole day now, basically, I tried implementing client component in a personal project but I am having trouble so I tried creating a fresh project but this is still the case. 

I tried removing the body element from the layout.tsx while retaining the children and it somehow removed the error I am getting and it rendered what I expected: 
![image](https://github.com/vercel/next.js/assets/81552056/00e029a0-5ec6-411e-a86f-701b78ad9597)

However, when I try clicking the button, nothing happens. so I checked the EventListeners via Chrome's Dev Tools and here is what I see:
![image](https://github.com/vercel/next.js/assets/81552056/97b0ccc8-b799-4171-82c9-fc68d189924a)

I am not familiar yet in checking event listeners but if I am not mistaken, shouldn't I see the a click event listener here for the button I created? 

I tried both Yarn and NPM but still the same issue. I tried both create-next-app and manually installing the packages but still the same issue.

Here's what the package.json of this current fresh project I made:
```
{
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "eslint": "^8.56.0",
    "next": "^14.0.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^5.3.3"
  },
  "devDependencies": {
    "@types/node": "20.11.5",
    "@types/react": "18.2.48"
  }
}
```

Please help! I literally read a bunch (20+) threads regarding client components not working and nothing seems to be working. I tried changing versions of node and downgrading/upgrading versions of NextJS but I still don't know what's causing this.

### Additional Information

Edition	Windows 11 Home Single Language
Version	22H2
OS build	22621.3007

Ran on local machine using yarn dev and npm run dev

**Comment by ananuness** [comment]
Hi Cedric!! Did you find what was the problem? Because I followed your same path and I didn't get any errors. The difference is that I used `next@latest` (14.1.0), also, I don't see any problem in your code :confused:

**Reply by cedricmarcelino** [reply]
I just gave up really. What I did was just use the page router instead of app router as I've tried everything I can and read a bunch of issue posts related to this one. When I switch to page router with the same versions I mentioned, it worked without any issue. It seems that the app router is not consistent yet? I'm not sure, I have read a lot about this kind of issue. See this https://github.com/vercel/next.js/discussions/43577

**Reply by cedricmarcelino** [reply]
https://github.com/vercel/next.js/discussions/43577#discussioncomment-7381848 This is what I feel inside.

**Reply by ananuness** [reply]
Wow, I knew that the app router has its problems, but not at this level, very strange for it to happen in such a specific way. And I laughed so hard at this thread comment lmao 😭😭😭😭

**Reply by cedricmarcelino** [reply]
Yeah I am not sure. I just didn't have the time to debug it more, it's kind of draining, but thank you for trying to help though! Cheers!
